zrml/prediction-markets/Cargo.toml

```
@@ -24,18 +28,20 @@ std = [
     'frame-support/std',
     'frame-system/std',
     'orml-traits/std',
-    'pallet-timestamp/std',
     'parity-scale-codec/std',
     'sp-arithmetic/std',
+    'serde',
     'sp-core/std',
     'sp-io/std',
     'sp-runtime/std',
+    'sp-std/std',
     'zeitgeist-primitives/std',
-    'zrml-swaps/std'
+    'zrml-swaps/std',
 ]
+runtime-benchmarks = ["frame-benchmarking", "orml-tokens", "pallet-balances"]
```

```suggestion
runtime-benchmarks = [
    "frame-benchmarking",
    "frame-support/runtime-benchmarks",
    "frame-system/runtime-benchmarks",
    "orml-tokens",
    "pallet-balances"
]
```

```
cargo test --features std,runtime-benchmarks --manifest-path zrml/prediction-markets/Cargo.toml --no-default-features
```

```
error[E0407]: method `add` is not a member of trait `$crate::traits::SortedMembers`
```

```
error[E0046]: not all trait items implemented, missing: `successful_origin`
```

After fixing compilation errors, `benchmarks::benchmark_tests::test_benchmarks` fails for some reason.

---

@c410-f3r I have fixed this problem, but there is one point where the pallet behaves strange.
When the benchmark tests are invoked with the command you supplied, `deploy_swap_pool_for_market` with sufficient base currency ztg (max balance) and categorical asset is also called. But it exits with an error, "InsufficientBalance" for `Asset::Ztg`. I have checked that the account has enough base currency, but the pallet reads 0 Asset::Ztg. Seems like during the prediction market tests the swaps pallet is unable to associate Asset::Ztg with the base currency. When I deposit Asset::Ztg token to the calling account before calling `deploy_swap_pool_for_market` in the benchmark, everything works as expected.

---

The reason for this is because in the mock for tests and benchmarks we [use the Tokens pallet instead of the Currencies pallet](https://github.com/zeitgeistpm/zeitgeist/pull/82/files#diff-2f51b5c84bbe63c6ee3df0a2b58d64c3fcc0cd5583ea23d2fb5b77c7464db9e5L172). We should use the Currencies pallet like how it's done in the the Swaps mock, [here](https://github.com/zeitgeistpm/zeitgeist/blob/main/zrml/swaps/src/mock.rs#L73) (we alias Shares to the Currencies pallet).

---

@lsaether done